operations/helm/charts/mimir-distributed/templates/tokengen/tokengen-job.yaml

```diff
@@ -65,32 +69,60 @@ spec:
           {{- if or .Values.global.extraEnv .Values.tokengenJob.env }}
           env:
             {{- with .Values.global.extraEnv }}
-              {{ toYaml . | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
             {{- end }}
             {{- with .Values.tokengenJob.env }}
-              {{ toYaml . | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
             {{- end }}
           {{- end }}
           {{- if or .Values.global.extraEnvFrom .Values.tokengenJob.extraEnvFrom }}
           envFrom:
             {{- with .Values.global.extraEnvFrom }}
-              {{- toYaml . | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
             {{- end }}
             {{- with .Values.tokengenJob.extraEnvFrom }}
-              {{- toYaml . | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
             {{- end }}
           {{- end }}
           securityContext:
             {{- toYaml .Values.tokengenJob.containerSecurityContext | nindent 12 }}
+      containers:
```

Comment:
is there a way to make this somehow conditional? Otherwise, this will add an additional image to GEM installations. I think some people running GEM are very frugal for what images they allow pulled.

Maybe we can have a regular container which polls for the token in the shared location and _then_ uploads it. That way we don't deploy a new image only to `echo "skipping creating secret"`

Reply:
This has been changed to new behavior:

If storeTokenInSecret is true, it runs tokengen as an initContainer and kubectl as a container.  If storeTokenInSecret is false, it will run tokengen as a container and omit kubectl image altogether.